src/components/Checklist.js

```diff
@@ -0,0 +1,80 @@
+import React from 'react'
+import _ from 'lodash'
+
+// Stateless, functional components. It is their responsibility to tranform
+// data for display. They should be designed so they will not raise an error,
+// and they must be contained by ErrorBoundaries if they might raise.
+const Header = ({title}) => <h2>{title}</h2>
+const className = (item) => item.checked ? 'item checked' : 'item unchecked'
+
+const Title = (ticketId) => <span>
+  Checklist {ticketId ? ` for ${ticketId}` : ''}
+  <a
+    style={{ margin: '0 0 0 10px' }}
+    href='.'
+  >
+    🔙
+  </a>
+</span>
+
+class Checklist extends React.Component {
+  constructor(props) {
+    super(props)
+
+    // This binding is necessary to make `this` work in the callback
+    this.toggleCheck = this.toggleCheck.bind(this);
+  }
+
+  toggleCheck = (ev) => {
+    const isChecked = () => ev.currentTarget.classList.contains('checked')
+    const content   = () => ev.currentTarget.textContent
+    this.itemToUpdate = { text: content(), checked: !isChecked() }
+    ev.preventDefault()
+    this.setState((pastState) => {
+      let index = _.findIndex(
+        pastState.items,
+        (item) => item.text === this.itemToUpdate.text
+      )
+
+      pastState.items[index] = this.itemToUpdate
+      localStorage.setItem('checklist', JSON.stringify(pastState))
+      return pastState
+    })
+  }
+
+  render() {
+    return <div id='checklist'>
+      <Header title={Title()} />
+      <ul>
+        {
+          _.chain(this.state && this.state.items).
```

Comment:
I'm not sure why you need lodash or `_.chain` here. Regular ES works just fine, is shorter, and IMO is more clear:

```
this.state.items.map((item, index) =>
  <li
    style={{listStyle: "none"}}
    className={className(item)}
    onClick={this.toggleCheck}
    key={ index }>{ item.text }
  </li>)
```